code/modules/projectiles/ammunition.dm

```diff
@@ -58,3 +59,17 @@
 				to_chat(user, "<span class='warning'>You fail to collect anything!</span>")
 	else
 		return ..()
+
+/obj/item/ammo_casing/throw_impact(atom/A)
+	if(heavy_metal)
+		bounce_away(FALSE, NONE)
+	. = ..()
+
+/obj/item/ammo_casing/proc/bounce_away(still_warm = FALSE, delay = 3)
+	SpinAnimation(10, 1)
+	update_icon()
+	var/turf/T = get_turf(src)
+	if(T && (isfloorturf(T) || isindestructiblefloor(T) || isclosedturf(T)))

```

Comment:
I believe explicit skip on space/vacuum would be better here in case we add more non-floors in the future.